### PR TITLE
Don't throw errors if error request does no contain json errros

### PR DESCRIPTION
**Pull request by grahamgilchrist**
This fixes a fatal error, when the request fails, but there are no errors in the response. This can legitimately be caused by several things:
* Server 500 error
* Offline functionality

### CODE REVIEW

**Comment by grahamgilchrist** [comment]
@incuna/js Pls review/merge

**Comment by wytrych** [comment]
Changelog?

**Comment by grahamgilchrist** [comment]
Added changelog

File: changelog.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 This project uses Semantic Versioning (2.0).
 
 ### UPCOMING
+* Avoid fatal error when api promise fails butt here are no errors
```
Comment (wytrych) [inline changelog.md]: Rude!

Comment (grahamgilchrist) [inline changelog.md]: lol. err.. fixing

File: web/app/scripts/compiled-es5/services/fieldsets-parser.js [hunk 2]
```diff
@@ -57,14 +57,16 @@ _module.service(moduleProperties.componentName, [function () {
     };
 
     this.addFieldErrors = function (fields, errors) {
-        fields.forEach(function (fieldset) {
-            var key = fieldset.templateOptions.id;
-            var fieldsetErrors = errors[key];
-            fieldset.fieldGroup.forEach(function (field) {
-                var fieldError = fieldsetErrors && fieldsetErrors[field.key];
-                field.templateOptions.fieldOptions.errors = fieldError;
+        if (fields && errors) {
```
Comment (henrahmagix) [inline web/app/scripts/compiled-es5/services/fieldsets-parser.js]: Fields and errors, fields and errors, I gotta have me my fields and errors

**Pull request merged**